src/query/api/v1/handler/prom/read_test.go

```diff
@@ -144,14 +145,14 @@ func TestPromReadHandlerInvalidQuery(t *testing.T) {
 	require.Equal(t, http.StatusBadRequest, recorder.Code)
 }
 
-func TestPromReadHandlerExecuteInvalidParamsError(t *testing.T) {
+func TestPromReadHandlerExecutePromError(t *testing.T) {
 	setup := setupTest(t)
 	setup.queryable.selectFn = func(
 		sortSeries bool,
 		hints *promstorage.SelectHints,
 		labelMatchers ...*labels.Matcher,
 	) (promstorage.SeriesSet, promstorage.Warnings, error) {
-		return nil, nil, xerrors.NewInvalidParamsError(fmt.Errorf("user input error"))
```

Comment:
was this test not realistic because prom wouldn't return an invalid params error?

Reply:
it could from the m3 storage layer.

http -> prom -> m3 storage return InvalidParamsErr

I guess I could add a third test that shows InvalidParamsErr from m3 storage is still a 4xx. i'll do that.

Reply:
time for test cases!